client/scala/src/main/scala/ai/verta/repository/PathList.scala

```diff
@@ -0,0 +1,15 @@
+package ai.verta.repository
+
+/** A class to represent a path in the list of strings format
+ *  @param components a list of components of the path
+ */
+final case class PathList(val components: List[String]) {
```

Comment:
Final? Did you mean sealed? What is the limitation you had in mind?

Reply:
I read somewhere that if I don't make case class final, user can extend it which leads to weird equality behavior. Probably overkill though. I'll remove it.

Reply:
Removed.

Reply:
Ah, ok. `final` is ok here indeed. You can keep or remove.